src/charm.py

```diff
@@ -129,6 +133,26 @@ def __init__(self, *args):
         # Relations hooks
         self.framework.observe(self.on.mme_relation_changed, self._mme_relation_changed)
 
+        # Lte core interface
+        self.core_requirer = CoreRequires(self, "lte-core")
+        self.framework.observe(
+            self.core_requirer.on.lte_core_available,
+            self._on_lte_core_available,
+        )
+
+    def _on_lte_core_available(self, event: CoreAvailableEvent) -> None:
+        mme_addr = event.mme_ipv4_address
+        if not is_ipv4(mme_addr):
+            return
```

Comment:
There's already a check on the lib side (iirc we'll also fire a custom event when IP is not valid). Do we need this check also here?

Reply:
Yeah, you recall correctly. Dropped.